src/Microsoft.QualityGuidelines.Analyzers/Core/MarkMembersAsStatic.Fixer.cs

```diff
@@ -22,9 +29,33 @@ public sealed override FixAllProvider GetFixAllProvider()
 
         public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
         {
-            // This is to get rid of warning CS1998, please remove when implementing this analyzer
-            await Task.Run(() => { }).ConfigureAwait(false);
-            throw new NotImplementedException();
+            SyntaxNode root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            SyntaxNode node = root.FindNode(context.Span);
+            var model = await context.Document.GetSemanticModelAsync(context.CancellationToken).ConfigureAwait(false);
+            var symbol = model.GetDeclaredSymbol(node, context.CancellationToken);
+
+            context.RegisterCodeFix(new MarkMembersAsStaticAction(
+                    string.Format(MicrosoftQualityGuidelinesAnalyzersResources.MarkMembersAsStaticCodeFix, symbol.Name),
+                    async ct => await RemoveStaticModifier(context.Document, root, node, ct).ConfigureAwait(false)
+                ), context.Diagnostics.Single());
+        }
+
+        private Task<Document> RemoveStaticModifier(Document document, SyntaxNode root, SyntaxNode node, CancellationToken ct)
+        {
+            ct.ThrowIfCancellationRequested();
+            var syntaxGenerator = SyntaxGenerator.GetGenerator(document);
+            var madeStatic = syntaxGenerator.WithModifiers(node, DeclarationModifiers.Static);
```

Comment:
ah, now I know what this fixer is trying to do. I think this require quite complex refactoring rather than simple adding static. I think we have 3 options.

1. do find all reference and make sure there is no reference to it and offer the fix (this most likely work for all small solution since FAR will be faster enough and fixer offer is interactive enough so that when user finish a method, the fix is offered for the method user is currently on)

2. implement the actual make member static refactoring service. and fix up definition and all references.

3. just add static and let user deal with errors.

for now, I would prefer 1) approach.